src/otx/algorithms/detection/adapters/mmdet/models/detectors/mean_teacher.py

```diff
@@ -0,0 +1,267 @@
+"""UnbiasedTeacher Class for mmdetection detectors."""
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import copy
+import functools
+
+import numpy as np
+import torch
+from mmdet.core import bbox2result, bbox2roi
+from mmdet.core.mask.structures import BitmapMasks
+from mmdet.models import DETECTORS, build_detector
+from mmdet.models.detectors import BaseDetector
+
+from otx.algorithms.common.utils.logger import get_logger
+
+from .sam_detector_mixin import SAMDetectorMixin
+
+logger = get_logger()
+
+# TODO: Need to fix pylint issues
+# pylint: disable=abstract-method, too-many-locals, unused-argument
+
+
+@DETECTORS.register_module()
+class MeanTeacher(SAMDetectorMixin, BaseDetector):
+    """Mean teacher framework for detection and instance segmentation."""
```

Comment:
Link for paper of this algorithm might help advanced users.

Reply:
There is no link to paper. It is enhanced version of UnbiasedTeacher with EMA for student, using losses for OD and enabling mask loss upon that. So, It is more like general idea of MT used in Semi-SL field